src/pages/MemberCollectionAdminPage/MemberCollectionAdminPage.tsx

```diff
@@ -115,25 +118,170 @@ const MemberCollectionAdminPage: React.FC = () => {
+      )}
+
+      {enabledModules.permission_group && permissions.MEMBER_PERMISSION_GROUP_SELECT ? (
+        <PermissionGroupsDropDownSelector fieldFilter={fieldFilter} onFiledFilterChange={setFieldFilter} />
+      ) : null}
+    </div>
+  )
+}
+
+const MemberImportExportControlPanel: React.FC<{
+  permissions: { [key: string]: boolean }
+  enabledModules: { [key: string]: boolean | undefined }
+  fieldFilter: FiledFilter
+  exportImportVersionTag: boolean
+  appId: string
+  sortOrder: {
+    createdAt: SortOrder
+    loginedAt: SortOrder
+    consumption: SortOrder
+  }
+  visibleColumnIds: string[]
+  columns: { key: string; title: string }[]
+}> = ({
+  permissions,
+  enabledModules,
+  fieldFilter,
+  exportImportVersionTag,
+  appId,
+  sortOrder,
+  visibleColumnIds,
+  columns,
+}) => {
+  return (
+    <>
+      <div className="mr-2">{permissions.MEMBER_CREATE && <MemberCreationModal />}</div>
+      <div className="mr-2">
+        {exportImportVersionTag ? (
+          <MemberImportModal />
+        ) : (
+          <OldMemberImportModal /> // TODO: remove this after new export import completed
+        )}
+      </div>
+      {enabledModules.member_info_export ? (
+        exportImportVersionTag ? (
+          <MemberExportModal appId={appId} filter={fieldFilter} sortOrder={sortOrder} />
+        ) : (
+          <OldMemberExportModal // TODO: remove this after new export import completed
+            appId={appId}
+            visibleFields={visibleColumnIds}
+            columns={columns}
+            filter={fieldFilter}
+            sortOrder={sortOrder}
+          />
+        )
+      ) : null}
+    </>
+  )
+}
+
+export const MemberFieldFilter: React.FC<{
+  allColumns: ({
+    id: string
+    title: string
+  } | null)[]
+  visibleColumnIds: string[]
+  setVisibleColumnIds: (value: string[]) => void
+}> = ({ allColumns, visibleColumnIds, setVisibleColumnIds }) => {
+  const { formatMessage } = useIntl()
+  return (
+    <Popover
+      trigger="click"
+      placement="bottomLeft"
+      content={
+        <StyledOverlay>
+          <OverlayTitle>{formatMessage(memberMessages.label.fieldVisible)}</OverlayTitle>
+          <Checkbox.Group value={visibleColumnIds} onChange={value => setVisibleColumnIds(value as string[])}>
+            <FilterWrapper>
+              {allColumns.map(column =>
+                column ? (
+                  <Checkbox key={column.id} value={column.id}>
+                    {column.title}
+                  </Checkbox>
+                ) : null,
+              )}
+            </FilterWrapper>
+          </Checkbox.Group>
+        </StyledOverlay>
+      }
+    >
+      <StyledButton type="link" icon={<Icon component={() => <TableIcon />} />}>
+        {formatMessage(memberMessages.label.field)}
+      </StyledButton>
+    </Popover>
+  )
+}
+
+export const MemberCollectionBlock: React.VFC<
   Pick<AppProps, 'enabledModules' | 'settings'> & {
     currentUserRole: string
     appId: string
```

Comment:
Is it unnecessary to be the props cause it's a global variable?

Reply:
Do you mean useContext?
It is indeed possible to get the value from useContext without using props, but this is the original way of writing it. I think it would be better to adjust it in refactor PR.